ui/app/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js

```diff
@@ -12,63 +12,69 @@ export default class PermissionPageContainerContent extends PureComponent {
     selectedPermissions: PropTypes.object.isRequired,
     permissionsDescriptions: PropTypes.object.isRequired,
     onPermissionToggle: PropTypes.func.isRequired,
-    selectedAccount: PropTypes.object,
+    selectedIdentities: PropTypes.array,
     redirect: PropTypes.bool,
     permissionRejected: PropTypes.bool,
   }
 
   static defaultProps = {
     redirect: null,
     permissionRejected: null,
-    selectedAccount: {},
+    selectedIdentities: [],
   }
 
   static contextTypes = {
     t: PropTypes.func,
   }
 
-  renderAccountInfo = (account) => {
+  renderRedirectVisual () {
+    const { permissionRejected, selectedIdentities, domainMetadata } = this.props
     return (
-      <div className="permission-approval-visual__account-info">
-        <div className="permission-approval-visual__account-info__label">
-          { account.label }
+      <div className="permission-approval-visual__redirect">
+        <IconWithFallBack icon={domainMetadata.icon} name={domainMetadata.name} />
+        <div className="permission-approval-visual__redirect-center">
+          { permissionRejected
+            ? <span className="permission-approval-visual__reject" ><i className="fa fa-times-circle" /></span>
+            : <span className="permission-approval-visual__check" />
+          }
+          <img className="permission-approval-visual__broken-line" src="/images/broken-line.svg" />
         </div>
-        <div className="permission-approval-visual__account-info__address">
-          { account.truncatedAddress }
+        <div className="permission-approval-visual__identicon-container">
+          <div className="permission-approval-visual__identicon-border">
+            <Identicon
+              className="permission-approval-visual__identicon"
+              address={selectedIdentities[0].address}
+              diameter={54}
+            />
+          </div>
         </div>
       </div>
     )
   }
 
+  renderNonRedirectVisual () {
```

Comment:
For some reason this is displayed with the icon overlapping the text when I run it locally

Reply:
Screenshot:
![broken-header](https://user-images.githubusercontent.com/2459287/78056058-e08fe900-735a-11ea-9233-348a7fed2a3b.png)

Reply:
I don't think this is an issue any more

![Screenshot from 2020-04-01 03-57-21](https://user-images.githubusercontent.com/7499938/78105943-ec1bf800-73cc-11ea-8497-7691c0e4fa39.png)